src/lib.rs

```diff
@@ -386,11 +405,65 @@ pub struct SentryCredential {
     pub host: String,
     pub project_id: String,
 }
+
 pub struct Sentry {
+    settings: Settings,
+    worker: Arc<SingleWorker<Event, SentryCredential>>,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Settings {
     server_name: String,
     release: String,
     environment: String,
-    worker: Arc<SingleWorker<Event, SentryCredential>>,
+    device: Device
+}
+
+pub struct SettingsBuilder {
+    server_name: Option<String>,
+    release: Option<String>,
+    environment: Option<String>,
+    device: Option<Device>
+}
+
+impl SettingsBuilder {
+    pub fn new() -> SettingsBuilder {
```

Comment:
Well I would prefer Settings that are set only once and no optionnal and default values as helper (just like Device). Settings are not changed at runtime...